### DISCUSSION TITLE
is there a way to find element by column value?

### DISCUSSION THREAD

**Discussion by xkungfu** [nlohmann/json#3179] [Q&A] [answered]
```
json j_patch = R"([
  { "op": "replace", "path": "/baz", "value": "boo" },
  { "op": "add", "path": "/hello", "value": ["world"] },
  { "op": "remove", "path": "/foo"}
])"_json;
```

how to find  { "op": "replace", "path": "/baz", "value": "boo" } and is index by "boo"?

return 0 and { "op": "replace", "path": "/baz", "value": "boo" }

**Comment by nlohmann** [comment, marked as answer]
Since the top level value is an array, you can use `j_patch[0]` to access the first element. Or do I misunderstand your question?

**Reply by xkungfu** [reply]
> 

sorry, I didn't describe it clearly.

my question should be:

how to search from a json object by a column name and value? the result return the element and the index number of the value.

the code should be looked like:

json j_patch = R"([
  { "op": "replace", "path": "/baz", "value": "boo" },
  { "op": "append", "path": "/new", "value": "boo" },
  { "op": "add", "path": "/hello", "value": ["world"] },
  { "op": "remove", "path": "/foo"}
])"_json;


```
res = search_element_by_column_and_value(j_patch, "value",  "boo")
std::cout << res.indexnumber;
std::cout << res.element;
```

the res output sholud like to:
```
0
1
  { "op": "replace", "path": "/baz", "value": "boo" },
  { "op": "append", "path": "/new", "value": "boo" },

```

**Reply by nlohmann** [reply]
You need to iterate the values of the array (like `for (auto i = 0; i < j_patch.size(); ++i`) and check for the key like `j_patch[i]["value"] == "boo"`.

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?
